### ISSUE TITLE
Convert absolute paths to new relatives with "../../"

### ISSUE THREAD

**Issue by GulinSS** [michaelzoidl/babel-root-import#14]
It makes plugin usable for these cases:
1. After `babel --out-dir dist app`
2. At webpack/browserify environment

**Comment by michaelzoidl** [comment]
You mean instead of converting it to the absolute root path (from `process.cwd)` converting it to the right order of `../../ etc`?

**Comment by GulinSS** [comment]
yes

**Comment by michaelzoidl** [comment]
hehe.. i had already the same thoughts a few weeks ago... i think it's time to update the logic (beside i think thats a way cleaner/better solution for this plugin).

But i think i new a few days to update the plugin since this is a private side-project...

**Comment by GulinSS** [comment]
It would be perfect to have this feature. It allows use plugin more widely.

**Comment by iltaen** [comment]
+1

**Comment by bambriy** [comment]
+1

**Comment by igor-slotin** [comment]
+1

**Comment by jilt1015** [comment]
+1

**Comment by michaelzoidl** [comment]
Ok, i created a new repo for this issue, the part where the module calculates the relative backward path already works, hope i make the other part also today.

Repo: https://github.com/michaelzoidl/dir-distance

**Comment by wiredearp** [comment]
:+1:

**Comment by GulinSS** [comment]
> https://github.com/michaelzoidl/dir-distance
I see 404 :-(

**Comment by michaelzoidl** [comment]
Yap i deleted it yesterday, since the target from the `dir-distance` is the same like the already provided `path.relative()` function from the native path module... 
Update is coming :) :+1:

**Comment by danawoodman** [comment]
Looking forward to this change, is there an ETA when a new release will be published?

**Comment by mikepombal** [comment]
Also looking forward to this change :)

**Comment by danawoodman** [comment]
@michaelzoidl do you need any help getting that feature added? It would allow me to use it on a few projects that use babel to build the proejct :smile:

**Comment by michaelzoidl** [comment]
A few days i had time to find a solution, with `path.resolve(pathA,pathB)` it's really simple to get the relative path - the problem is: i don't find the file-origin-path from the babel-plugin-api. 
But guys, i didn't forget you. :/ Hope i find soon a solution, sorry.

**Comment by danawoodman** [comment]
Thanks @michaelzoidl let us know if we can help in some way as with this feature this project 100% solves the `../../../` require problem for me (and I'm sure many others who use babel)

**Comment by GulinSS** [comment]
@michaelzoidl any news?

**Comment by michaelzoidl** [comment]
I asked in the `bable-handbook` repo if there is some API for this, when i get the answer i finish the implementation. Link to Issue: https://github.com/thejameskyle/babel-handbook/issues/63

**Comment by DimitriGilbert** [comment]
after browsing through babel source code trying to make sense of what i read, i think, maybe (MAYBE) you could access the file name through `state.opts.filename`.
what makes me say that is located in [here](https://github.com/babel/babel/blob/master/packages/babel-cli/src/babel/util.js) between line 36 and 60 (i'll add comment of what i understood) : 
```javascript
// the function called to transform the code
export function transform(filename, code, opts) {
  //merging existing options with given options
  opts = _.defaults(opts || {}, index.opts);
  // adding filename to the options
  opts.filename = filename;
  opts.ignore = null;
  opts.only = null;

  // run babel transform
  let result = babel.transform(code, opts);
  result.filename = filename;
  result.actual = code;
  return result;
}

// function called in ../file.js::123 to create the output
export function compile(filename, opts) {
  try {
    // get the content of the file
    let code = fs.readFileSync(filename, "utf8");
    // run the function above
    return transform(filename, code, opts);
  } catch (err) {
    if (commander.watch) {
      console.error(toErrorStack(err));
      return { ignored: true };
    } else {
      throw err;
    }
  }
}
```
I found no reference of it in the handbook though so it might be bogus....
anyway i hope it helps.

**Comment by danawoodman** [comment]
Don't mean to pester but any timeline on this?

**Comment by tonyhb** [comment]
Use this plugin instead: https://github.com/tleunen/babel-plugin-module-alias

**Comment by DimitriGilbert** [comment]
i switched to require statement using the "paths" configuration in browserify.

**Comment by wiredearp** [comment]
(there's also https://github.com/per2plex/babel-project-relative-import which retains the configurable ~tilde syntax because it is "highly inspired" by this project)

**Comment by danawoodman** [comment]
@michaelzoidl are you planning on incorporating that change or should we move over to the above mentioned package?

**Comment by michaelzoidl** [comment]
Hi guys, sorry for the extrem late answer, so my current problem is that i'm fully booked (i'm a freelancer).
So i've found a solution but have not found the time to implement this feature :( 

If you need the feature _now_ i would say move on to the other project - if you're interested to use still this project but want this feature a PR is always welcome.

Sorry again :/

**Comment by danawoodman** [comment]
No worries, @michaelzoidl I get it (as I'm a freelancer myself). For now I'll use another library but will keep an eye here

**Comment by smeijer** [comment]
I'm glad that your business is going good. But I am curious, any update on this?

**Comment by danawoodman** [comment]
@smeijer you might want to checkout https://github.com/tleunen/babel-plugin-module-alias which solves the problem for me (it is actually a bit more robust)

**Comment by nescalante** [comment]
Hi all, I opened a PR that fixes this problem.

**Comment by michaelzoidl** [comment]
:) guys, special thanks to @nescalante we can finally close this... 
Released in version `4.1.0`

**Comment by nescalante** [comment]
Thanks!
